src/data/repositories/utils/builders/UploadsFormDataBuilder.ts

```diff
@@ -0,0 +1,24 @@
+import { ImportSummary, ImportSummaryErrors } from "../../../../domain/entities/data-entry/ImportSummary";
+import { BrowserUploadsFormDataBuilder } from "./BrowserUploadsFormDataBuilder";
+import { NodeUploadsFormDataBuilder } from "./NodeUploadsFormDataBuilder";
+
+import type FormDataNode from "form-data";
+
+export type UploadsFormData = FormData | FormDataNode;
+
+export interface UploadsFormDataBuilder {
+    createAsyncImportSummariesFormData(importSummaries: ImportSummary[]): UploadsFormData;
+    createImportSummaryFormData(importSummary: ImportSummaryErrors): UploadsFormData;
+}
+
+export function getUploadsFormDataBuilder(runtime: "node" | "browser"): UploadsFormDataBuilder {
+    switch (runtime) {
```

Comment:
This violates the inversion of dependencies principle. A repository should be agnostic of the environment (browser or node).

A repository should not create the builder dependency. A better option is to decide in the composition root and inject the builder in the constructor